plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/nessie/TestIcebergNessieCatalogConnectorSmokeTest.java

```diff
@@ -42,8 +43,10 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.jupiter.api.Assumptions.abort;
 import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.SAME_THREAD;
 
 @TestInstance(PER_CLASS)
+@Execution(SAME_THREAD)
```

Comment:
Nessie (Iceberg) supports concurrent commits via Iceberg framework itself (via `BaseTransaction.commitSimpleTransaction`. But the chances of flakiness or collision is more if we just run the testcase with default commit retry count (4). If we increase the retry count, test cases can pass. But if we add more new testcases, we may need to increase the retry count some more. Hence, made it single thread.

Reply:
Thanks for clarifying @ajantha-bhat - do we maybe add this as javadoc for this test then?

Reply:
I couldn't reproduce these flakiness locally. These fixes were done based on code review. 
Let us keep it single threaded for few versions to see if that was the cause. (most likely). 

How many concurrent inserts does Trino supports per Iceberg table? 

point 3) from this blog talks more about Iceberg concurrency
https://medium.com/@ajanthabhat/how-not-to-use-apache-iceberg-046ae7e7c884

Reply:
> How many concurrent inserts does Trino supports per Iceberg table?

Running tests concurrently doesn't change the number of concurrent inserts to an Iceberg tables, as distinct tests don't **write** to same Iceberg table. Why would they?

Reply:
I am suspecting two testcases might be using same table name ( I saw that we have random suffix in the code, but still it might have happened since it is not failing everytime). 

Let us wait for 2-3 weeks to see if the Nessie testcases fails again or not with that fix merged. 
I will support in analyzing the failure if it happens again. But hard to come to conclusions with concurrency enabled. So, single thread will definitely helps in narrowing down the problem.

Reply:
> I am suspecting two testcases might be using same table name

for read-only usages like `nation` or `orders` -- yes

for modifications -- not possible, since every such test creates a new table, so one of them should fail at CREATE time

> Let us wait for 2-3 weeks to see if the Nessie testcases fails again or not with that fix merged.

First and foremost, per my current understanding, this PR was not a fix, but just hiding the problem. This is what we're discussing here. Even if it fixes tests, it doesn't fix the production issue that seems to be indicated by those tests being flaky. What if, for example, Nessie catalog does not support concurrent commits to distinct tables?

Reply:
Let me add the concurrent commit testcases.